Block/Checkout/Onepage/Success/TescoAdditionalInformation.php

```diff
@@ -31,23 +33,21 @@ protected function _toHtml()
     {
         $paymentData = $this->_checkoutSession->getLastRealOrder()->getPayment()->getData();
         if ($paymentData['additional_information']['payment_type'] !== 'bill_payment_tesco_lotus') {
-            return '';
+            return parent::_toHtml();
         }
-        $tescoCodeImageUrl =  $paymentData['additional_information']['barcode'];
+        $barcode =  $paymentData['additional_information']['barcode'];
 
-        if (!$tescoCodeImageUrl) {
-            return '';
+        if (!$barcode) {
+            return parent::_toHtml();
         }
-        
+
         $orderCurrency = $this->_checkoutSession->getLastRealOrder()->getOrderCurrency()->getCurrencyCode();
-        
-        $this->addData(
-            [
-                'tesco_code_url' => $tescoCodeImageUrl,
-                'order_amount' => number_format($paymentData['amount_ordered'], 2) .' '.$orderCurrency
-            ]
-        );
 
+        $this->addData([
+            'tesco_code_image' => $barcode,
+            'order_amount' => number_format($paymentData['amount_ordered'], 2) .' '.$orderCurrency
+        ]);
+        
```

Comment:
Just in case, if you are looking for refactoring this method. It could go with

```php
protected function _toHtml()
{
    $paymentData    = $this->_checkoutSession->getLastRealOrder()->getPayment()->getData();
    $additionalInfo = $paymentData['additional_information'];

    if ($additionalInfo['payment_type'] === 'bill_payment_tesco_lotus' && $additionalInfo['barcode']) {
        $orderCurrency = $this->_checkoutSession->getLastRealOrder()->getOrderCurrency()->getCurrencyCode();

        $this->addData([
            'tesco_code_image' => $additionalInfo['barcode'],
            'order_amount' => number_format($paymentData['amount_ordered'], 2) . ' ' . $orderCurrency
        ]);
    }

    return parent::_toHtml();
}
```

Would be more cleaner as every conditions will be returning `return parent::_toHtml();`.